CMakeLists.txt

```diff
@@ -28,6 +28,14 @@ endif()
 if(CMAKE_SYSTEM_PROCESSOR)
 	if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "^arm")
 		set(ARM ON)
+			if(UNIX AND NOT APPLE)
+				execute_process(COMMAND cat /proc/cpuinfo OUTPUT_VARIABLE OUTSTR)
+				string(FIND "${OUTSTR}" "ODROID-XU" pos)
+				if(pos GREATER_EQUAL 0)
+					add_definitions(-mfloat-abi=hard -marm -mtune=cortex-a15.cortex-a7 -mcpu=cortex-a15 -mfpu=neon-vfpv4 -fomit-frame-pointer -ffast-math -DARM_NEON -DODROID)
```

Comment:
A few things here:

1. I don't think `-DARM_NEON` is needed at all.  The compiler defines `__ARM_NEON__` which is what we look for.
2. I'd say we should just add `fast-math` for all gcc/clang, or at least all armv7.  No reason to make it ODROID specific.
3. I know there are a couple armv7 devices lacking NEON, but my preference would be to add `-mfpu=neon-vfpv4` for all armv7 as well.  We could add a cmake option for disabling it if necessary.

Can you run:
```
cmake -E environment
```

Does anything there already contain something that says "hard" or "hardfp"?  It'd be nice to detect hardfp for more than just ODROID.

-[Unknown]

Reply:
yes and __ARM_NEON as soon you define ``-mfpu=neon`` that is a left over from the past, I will skip it ;)